packages/plugin-commands-setup/test/setup.win32.test.ts

```diff
@@ -0,0 +1,171 @@
+
+let originalShell: string | undefined
+let originalPlatform = ''
+
+beforeAll(() => {
+  originalShell = process.env.SHELL
+  originalPlatform = process.platform
+
+  process.env.SHELL = ''
+  Object.defineProperty(process, 'platform', {
+    value: 'win32',
+  })
+})
+
+afterAll(() => {
+  process.env.SHELL = originalShell
+  Object.defineProperty(process, 'platform', {
+    value: originalPlatform,
+  })
+})
+
+test('Win32 registry environment values could not be retrieved', async () => {
+  execa['mockResolvedValue']({
+    failed: true,
+  })
+
+  const output = await setup.handler({
+    pnpmHomeDir: __dirname,
+  })
+
+  expect(output).toContain('Win32 registry environment values could not be retrieved')
+})
+
+test('Environment PATH is not configured correctly', async () => {
+  execa['mockResolvedValueOnce']({
+    failed: false,
+    stdout: 'SOME KIND OF ERROR OR UNSUPPORTED RESPONSE FORMAT',
+  }).mockResolvedValue({
+    failed: true,
+  })
+
+  const output = await setup.handler({
+    pnpmHomeDir: __dirname,
+  })
+
+  expect(output).toContain('Current PATH is not set. No changes to this environment variable are applied')
+})
+
+test('Environment PATH is empty', async () => {
+  execa['mockResolvedValueOnce']({
+    failed: false,
+    stdout: `
+HKEY_CURRENT_USER\\Environment
+    Path    REG_EXPAND_SZ    
+`,
+  }).mockResolvedValue({
+    failed: false,
+  })
+
+  const output = await setup.handler({
+    pnpmHomeDir: __dirname,
+  })
+
+  expect(output).toContain('Current PATH is empty. No changes to this environment variable are applied')
+})
+
+test('Successful first time installation', async () => {
+  execa['mockResolvedValueOnce']({
+    failed: false,
+    stdout: `
+HKEY_CURRENT_USER\\Environment
+    Path    REG_EXPAND_SZ    %USERPROFILE%\\AppData\\Local\\Microsoft\\WindowsApps;%USERPROFILE%\\.config\\etc;
+`,
+  }).mockResolvedValueOnce({
+    failed: false,
+    stdout: 'PNPM_HOME ENV VAR SET',
+  }).mockResolvedValueOnce({
+    failed: false,
+    stdout: 'PATH UPDATED',
+  }).mockResolvedValue({
+    failed: true,
+    stderr: 'UNEXPECTED',
+  })
+
+  const output = await setup.handler({
+    pnpmHomeDir: __dirname,
+  })
+
+  expect(output).toContain(`Setting 'PNPM_HOME' to value '${__dirname}`)
+  expect(output).toContain('Updating PATH')
+  expect(output).toContain('PNPM_HOME ENV VAR SET')
+  expect(output).toContain('PATH UPDATED')
+})
+
+test('PNPM_HOME is already set, but path is updated', async () => {
+  execa['mockResolvedValueOnce']({
+    failed: false,
+    stdout: `
+HKEY_CURRENT_USER\\Environment
+    PNPM_HOME    REG_EXPAND_SZ    .pnpm\\home
+    Path    REG_EXPAND_SZ    %USERPROFILE%\\AppData\\Local\\Microsoft\\WindowsApps;%USERPROFILE%\\.config\\etc;
+`,
+  }).mockResolvedValueOnce({
+    failed: false,
+    stdout: 'PATH UPDATED',
+  }).mockResolvedValue({
+    failed: true,
+    stderr: 'UNEXPECTED',
+  })
+
+  const output = await setup.handler({
+    pnpmHomeDir: __dirname,
+  })
+
```

Comment:
Probably also check that execa was called with the right args.

```
expect(execa).toBeCalledWith('reg', ....
```

Reply:
I enhanced tests with ```.toHaveBeenNthCalledWith```, however, because of mockup of ```execa``` module, I could not keep them in single test file. As far as I can tell, due to order of execution of tests, that check didn't work as expected.